### PR TITLE
Typecast XCOM response as per OSS PEP-563 type hint changes 

### LINKED ISSUE #658: Typecast XCOM response as per OSS PEP-563 type hint changes

### PR DESCRIPTION
**Pull request by rajaths010494**
mypy issue `task.output` returned type XComArg so mypy was throwing errors.
This PR type casts the XComArg to relevant datatypes to the operators/sensors needed.

closes #658

### CODE REVIEW

File: astronomer/providers/amazon/aws/example_dags/example_emr_eks_containers_job.py [hunk 1]
```diff
@@ -146,7 +146,7 @@ def create_emr_virtual_cluster_func(task_instance: Any) -> None:
     # [START howto_sensor_emr_job_container_async]
     emr_job_container_sensor = EmrContainerSensorAsync(
         task_id="emr_job_container_sensor",
-        job_id=run_emr_container_job.output,
+        job_id=str(run_emr_container_job.output),
```
Comment (phanikumv) [inline astronomer/providers/amazon/aws/example_dags/example_emr_eks_containers_job.py:149]: how / why was this working before and not now?

Comment (rajaths010494) [reply astronomer/providers/amazon/aws/example_dags/example_emr_eks_containers_job.py:149]: this still works but` EmrContainerSensorAsync` has attribute `job_id`  which accepts type `str` but the output type is `XComArg` now so mypy complains  `Argument "job_flow_id" to "EmrTerminateJobFlowOperator" has incompatible type "XComArg"; expected "str"  [arg-type]`.

Comment (rajaths010494) [reply astronomer/providers/amazon/aws/example_dags/example_emr_eks_containers_job.py:149]: Apply PEP-563 (Postponed Evaluation of Annotations) to core airflow (https://github.com/apache/airflow/pull/26290
https://github.com/apache/airflow/pull/26290)

**Review by bharanidharan14 — APPROVED** [review]

**Review by phanikumv — COMMENTED** [review]
PR title is not as per standards

File: astronomer/providers/amazon/aws/example_dags/example_emr_eks_containers_job.py (hunk 1 above)
Comment (pankajastro) [reply astronomer/providers/amazon/aws/example_dags/example_emr_eks_containers_job.py:149]: changes looking good to me

**Comment by rajaths010494** [comment]
Thanks @pankajastro .

**Pull request merged**